src/main/java/org/opensearch/knn/index/query/KNNQueryFactory.java

```diff
@@ -96,7 +95,7 @@ public static Query create(CreateQueryRequest createQueryRequest) {
                         .rescoreContext(rescoreContext)
                         .build();
             }
-            return isKnnQueryRewriteEnabled() ? new NativeEngineKnnVectorQuery(knnQuery) : knnQuery;
+            return createQueryRequest.getRescoreContext().isPresent() ? new NativeEngineKnnVectorQuery(knnQuery) : knnQuery;
```

Comment:
@jmazanec15 Please verify this condition is correct

Reply:
yeah this looks good